### PR TITLE
Added support for seeded content in local provider

### PR DESCRIPTION
**Pull request by Revolyssup**
**Description**

This PR adds out of the box support for filters and patterns built into the meshery server's file system, for local provider.

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@Revolyssup Heads up! Lint errors

**Comment by leecalcote** [comment]
Reminder: add labels to PRs when you can.

File: helpers/seed.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
+package helpers
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/gofrs/uuid"
+	models "github.com/layer5io/meshery/models"
+	"github.com/sirupsen/logrus"
```
Comment (DelusionalOptimist) [inline helpers/seed.go:10]: If we can, let's use meshkit logger here?
https://github.com/layer5io/meshkit/blob/master/logger/logger.go

File: helpers/seed.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+)
+
+type contentandName struct {
```
Comment (DelusionalOptimist) [inline helpers/seed.go:13]: Is this new struct needed? Can we use the ones for patterns and filters already defined in `models` package? Helps us keep uniformity and maintainability.
You can also then assign UUIDs while reading seed content from the filesystem.

File: helpers/seed.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}
+type Component string
+
+const (
+	Filter  Component = "FILTER"
+	Pattern Component = "PATTERN"
+)
+
+var seededUUIDs []uuid.UUID
+var localProvider *models.DefaultLocalProvider
```
Comment (DelusionalOptimist) [inline helpers/seed.go:25]: Any reason for not using the original reference to `DefaultLocalProvider` and creating this global variable? I think we can pass on that reference to other functions too and avoid this new var. :thinking:

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]

File: provider-ui/package-lock.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
```
Comment (DelusionalOptimist) [inline provider-ui/package-lock.json:5]: unintended changes here :smile:

File: helpers/seed.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
+package helpers
```
Comment (utkarsh-pro) [inline helpers/seed.go:1]: @Revolyssup why do any of this here? If we already are being explicit that this is a functionality of none provider then add a new method to `none` provider? This approach will result in the following:
1. You don't to keep a global variable around to keep track of the provider
2. You can use the meshkit logger

File: helpers/seed.go (hunk 2 above)
Comment (Revolyssup) [reply helpers/seed.go:13]: this is a redundant struct, removing it anyways

File: helpers/seed.go (hunk 3 above)
Comment (Revolyssup) [reply helpers/seed.go:25]: it was a mistake

File: helpers/seed.go (hunk 5 above)
Comment (Revolyssup) [reply helpers/seed.go:1]: @utkarsh-pro  Can I leave the helper function here? which fetched the files from file system and returns them as a string.
The function which is called can then be a method of local provider.

Comment (Revolyssup) [reply helpers/seed.go:1]: @utkarsh-pro  Then again, how do I keep track of the uuids of seeded components?

Comment (Revolyssup) [reply helpers/seed.go:1]: Actually cant leave helper functions there due to import cycle.

Comment (Revolyssup) [reply helpers/seed.go:1]: Can I add the SeededUUID's field in localProvider struct?

Comment (utkarsh-pro) [reply helpers/seed.go:1]: You don't have to keep track of the generated UUIDs in the provider. Here are rough function signatures:
```go
// cmd/main.go
// ...
seededUUIDs, err := provider.SeedContent()
// ...
<- c 
// Clean up
provider.CleanupSeeded(seededUUIDs)
```

**Comment by Revolyssup** [comment]
@utkarsh-pro  can you review the pr

File: cmd/main.go [hunk 6]
```diff
@@ -268,5 +269,7 @@ func main() {
 		}
 	}()
 	<-c
+	logrus.Info("Doing seeded content cleanup...")
+	lProv.CleanupSeeded(seededUUIDs)
```
Comment (utkarsh-pro) [inline cmd/main.go:273]: Do we actually need this cleanup?

Comment (Revolyssup) [reply cmd/main.go:273]: The end note was to keep it for now.
@leecalcote  can you confirm?

File: models/default_local_provider.go [hunk 7]
```diff
@@ -1104,3 +1157,37 @@ func genericHTTPApplicationFile(fileURL string) ([]MesheryApplication, error) {
 
 	return []MesheryApplication{af}, nil
 }
+func getSeededComponents(comp string, log logger.Handler) ([]string, []string, error) {
+	wd, err := os.UserHomeDir()
```
Comment (utkarsh-pro) [inline models/default_local_provider.go:1161]: Use meshkit's HomeDir method?

Comment (Revolyssup) [reply models/default_local_provider.go:1161]: cool

File: cmd/main.go (hunk 6 above)
Comment (utkarsh-pro) [reply cmd/main.go:273]: The advantage to have this is:
If the volume attached to meshery server is persisted across meshery restarts in those case we will end up with duplicate in the database.

**Comment by Revolyssup** [comment]
@utkarsh-pro  is this ready to merge?

**Comment by utkarsh-pro** [comment]
> @utkarsh-pro is this ready to merge?

Yep :+1:

**Review by utkarsh-pro — APPROVED** [review]
LGT

**Pull request merged**